docs/central-upgrade.rst

```diff
@@ -25,65 +26,342 @@ Upgrade notes
+
+   .. code-block:: console
+
+     $ docker --version && docker compose version
+
+   If you are using old versions, follow the instructions to install `Docker Engine <https://docs.docker.com/engine/install/ubuntu>`_ (not Desktop) for Ubuntu, the operating system we recommend and support. The instructions will help you setup the Docker ``apt`` repository and install the latest version of Docker Engine and Docker Compose.
+
+#. **Remove docker-compose.** You will be using ``docker compose`` from now on (the dash has been replaced with a space).
+
+   .. code-block:: console
+
+     $ rm -f `which docker-compose`
+
+#. **Migrate configuration customizations.** This will simplify future Central upgrades. First, check what files have been customized:
+
+   .. code-block:: console
+
+     $ git status
+
+   If you do not see any files listed with a `modified:` prefix, you can go on to the next step.
+
+   If you see files listed with a `modified:` prefix, follow instructions for each of them:
+
+   .. dropdown:: ``files/service/config.json.template``
+     :icon: file-code
+
+     #. See what changes have been made:
+
+        .. code-block:: console
+
+          $ git diff files/service/config.json.template
+
+        You will see additions in green with ``+`` prefixes. Copy those to a scratch file on your local computer. You will use this to copy your custom values into the new format.
+
+     #. Open the ``.env`` file for editing:
+
+        .. code-block:: console
+
+          $ nano .env
+
+     #. If you use a custom database server, you will see changes in the database section. Copy the values from that section to the ``.env`` file in the following format:
+
+        .. code-block:: bash
+
+          DB_HOST=my-db-host
+          DB_USER=my-db-user
+          DB_PASSWORD=my-db-password
+          DB_NAME=my-db-name
+
+        .. note::
+
+          If your password has special characters in it, you will need to put single quotes ( ``'`` and ``'``) around the password. Values without special characters do not need quotes around them.
+
+     #. If you use a custom email server, you will see changes in the email section. Copy those values to the ``.env`` file in the following format:
+
+        .. code-block:: bash
+
+          EMAIL_HOST=my-email-host
+          EMAIL_PORT=my-email-port
+          EMAIL_IGNORE_TLS=true-or-false
+          EMAIL_SECURE=true-or-false
+          EMAIL_USER=my-email-user
+          EMAIL_PASSWORD=my-email-password
+
+        .. note::
+
+          ``EMAIL_IGNORE_TLS`` should generally be set to ``false``. ``EMAIL_SECURE`` should generally be set to ``true`` if you use port 465 and to ``false`` for other ports.
+
+     #. Discard all ``files/service/config.json.template`` customizations. Make sure you have correctly copied all of them into ``.env``. You may want to keep the copy on your local computer until you have verified that everything works.
```

Comment:
Oh there's the message about making a backup, right after it said to get rid of it 😅 "You may want to keep the copy on your local computer until you have verified that everything works."

Reply:
"Make sure you have correctly copied all of them into" ... 
It's unclear how to copy other customizations into .env and which ones will be supported. You could say "Refer to [.env.template](https://github.com/getodk/central/blob/next/.env.template)  (but in master, not next) to see which things will be configured now in .env.

Reply:
`email.serviceAccount` --> `EMAIL_NO_REPLY` ?

Reply:
> Oh there's the message about making a backup

Ah! I think you may have missed the part about copying the diff lines into a file on your local computer? Maybe that needs to be clarified?

> email.serviceAccount --> EMAIL_NO_REPLY ?

Indeed, good catch!

Reply:
It's true, I did not read the sentence that said "Copy those to a scratch file on your local computer. You will use this to copy your custom values into the new format."  Though it is what I ended up doing.

Reply:
Oops! Ok, I changed the step to make it about copying since that's the important action. How does that look?